pkg/types/ccipocr3/interfaces.go

```diff
@@ -53,3 +58,15 @@ type ExecutePluginCodec interface {
 type MessageHasher interface {
 	Hash(context.Context, CCIPMsg) (Bytes32, error)
 }
+
+type HomeChainPoller interface {
+	GetChainConfig(chainSelector ChainSelector) (ChainConfig, error)
+	GetAllChainConfigs() (map[ChainSelector]ChainConfig, error)
+	// GetSupportedChainsForPeer Gets all chain selectors that the peerID can read/write from/to
+	GetSupportedChainsForPeer(id libocrtypes.PeerID) (mapset.Set[ChainSelector], error)
```

Comment:
How widely adopted is mapset? Generally I would avoid dependency types as part of a shared interface. Go uses `map[ChainSelector]bool` (or `map[ChainSelector]struct{}`) for sets.

Reply:
Right, this does not make sense to send over the wire. We could even return a plain `[]ChainSelector` and let the caller construct a set only if necessary.

Reply:
good question, I saw it used in a couple of places in CCIP. Didn't notice that it was that little though.